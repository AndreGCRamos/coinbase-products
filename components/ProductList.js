import * as React from 'react';
import { Dimensions, Text, View, StyleSheet, Image, FlatList } from 'react-native';

import { getAllStats } from '../assets/scripts/ProductListApi';

function Item({ item }) {
  return (
    <View style={styles.item}>
      <Text style={styles.id}>{item.id}</Text>
      <Text>Open: {item.open}</Text>
      <Text>High: {item.high}</Text>
      <Text>Low: {item.low}</Text>
      <Text>Volume: {item.volume}</Text>
    </View>
  );
}

export default class ProductList extends React.Component {
  constructor() {
    super();
    this.state = {
      productList: []
    };
  }

  componentDidMount() {
    getAllStats((stats) => {
      this.setState(prevState => ({ productList: [...prevState.productList, stats] }))
    })
  }

  render() {
    return (
      <View style={styles.container}>
        <FlatList
          data={this.state.productList}
          renderItem={({ item }) => <Item item={item} />}
          keyExtractor={product => product.id}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: Dimensions.get('window').width - 30,
  },
  item: {
    marginBottom: 24,
    padding: 10,
    backgroundColor: 'rgb(255, 255, 255)',
  },
  id: {
    fontSize: 14,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
