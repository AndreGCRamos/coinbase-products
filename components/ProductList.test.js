import 'react-native';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';

import ProductList from './ProductList';
import { dummyListSingle, dummyListMultiple } from '../assets/scripts/dummyData';

Enzyme.configure({ adapter: new Adapter() });

describe('<ProductList />', () => {
  it('renders', () => {
    const wrapper = mount(<ProductList />);
    wrapper.setState({ productList: dummyListSingle });
    expect(wrapper).toMatchSnapshot();
  });

  it('displays multiple items',() => {
    const wrapper = mount(<ProductList />);
    wrapper.setState({ productList: dummyListMultiple });
    const items = wrapper.find('Item');
    expect(items).toMatchSnapshot();
    expect(items.length).toBe(3);
  });
});

