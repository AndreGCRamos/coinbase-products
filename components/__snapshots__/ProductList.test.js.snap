// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> displays multiple items 1`] = `
Array [
  <Item
    item={
      Object {
        "high": "0.27933000",
        "id": "LTC-BTC",
        "last": "0.27020300",
        "low": "0.26416200",
        "open": "0.26416200",
        "volume": "883990.82612000",
        "volume_30day": "57077184.26522",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "marginBottom": 24,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "marginBottom": 24,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            LTC-BTC
          </Text>
        </Text>
        <Text>
          <Text>
            Open: 
            0.26416200
          </Text>
        </Text>
        <Text>
          <Text>
            High: 
            0.27933000
          </Text>
        </Text>
        <Text>
          <Text>
            Low: 
            0.26416200
          </Text>
        </Text>
        <Text>
          <Text>
            Volume: 
            883990.82612000
          </Text>
        </Text>
      </View>
    </View>
  </Item>,
  <Item
    item={
      Object {
        "high": "0.8786868868",
        "id": "ZRX-EUR",
        "last": "44.27020300",
        "low": "3.4422232",
        "open": "44.3343434",
        "volume": "7878987.82612000",
        "volume_30day": "44444444.333333",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "marginBottom": 24,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "marginBottom": 24,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            ZRX-EUR
          </Text>
        </Text>
        <Text>
          <Text>
            Open: 
            44.3343434
          </Text>
        </Text>
        <Text>
          <Text>
            High: 
            0.8786868868
          </Text>
        </Text>
        <Text>
          <Text>
            Low: 
            3.4422232
          </Text>
        </Text>
        <Text>
          <Text>
            Volume: 
            7878987.82612000
          </Text>
        </Text>
      </View>
    </View>
  </Item>,
  <Item
    item={
      Object {
        "high": "0.00688000",
        "id": "XLM-BTC",
        "last": "0.00670900",
        "low": "0.00662900",
        "open": "0.00674700",
        "volume": "22839.11587903",
        "volume_30day": "688557.00971328",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "marginBottom": 24,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "marginBottom": 24,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            XLM-BTC
          </Text>
        </Text>
        <Text>
          <Text>
            Open: 
            0.00674700
          </Text>
        </Text>
        <Text>
          <Text>
            High: 
            0.00688000
          </Text>
        </Text>
        <Text>
          <Text>
            Low: 
            0.00662900
          </Text>
        </Text>
        <Text>
          <Text>
            Volume: 
            22839.11587903
          </Text>
        </Text>
      </View>
    </View>
  </Item>,
]
`;

exports[`<ProductList /> renders 1`] = `
<ProductList>
  <View
    style={
      Object {
        "width": 720,
      }
    }
  >
    <View
      style={
        Object {
          "width": 720,
        }
      }
    >
      <FlatList
        data={
          Array [
            Object {
              "high": "0.27933000",
              "id": "BTC-XXX",
              "last": "0.27020300",
              "low": "0.26416200",
              "open": "0.26416200",
              "volume": "883990.82612000",
              "volume_30day": "57077184.26522",
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      >
        <VirtualizedList
          data={
            Array [
              Object {
                "high": "0.27933000",
                "id": "BTC-XXX",
                "last": "0.27020300",
                "low": "0.26416200",
                "open": "0.26416200",
                "volume": "883990.82612000",
                "volume_30day": "57077184.26522",
              },
            ]
          }
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <ScrollViewMock
            data={
              Array [
                Object {
                  "high": "0.27933000",
                  "id": "BTC-XXX",
                  "last": "0.27020300",
                  "low": "0.26416200",
                  "open": "0.26416200",
                  "volume": "883990.82612000",
                  "volume_30day": "57077184.26522",
                },
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <_class
              data={
                Array [
                  Object {
                    "high": "0.27933000",
                    "id": "BTC-XXX",
                    "last": "0.27020300",
                    "low": "0.26416200",
                    "open": "0.26416200",
                    "volume": "883990.82612000",
                    "volume_30day": "57077184.26522",
                  },
                ]
              }
              disableVirtualization={false}
              getItem={[Function]}
              getItemCount={[Function]}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onContentSizeChange={[Function]}
              onEndReachedThreshold={2}
              onLayout={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={50}
              stickyHeaderIndices={Array []}
              updateCellsBatchingPeriod={50}
              viewabilityConfigCallbackPairs={Array []}
              windowSize={21}
            >
              <RCTScrollView
                data={
                  Array [
                    Object {
                      "high": "0.27933000",
                      "id": "BTC-XXX",
                      "last": "0.27020300",
                      "low": "0.26416200",
                      "open": "0.26416200",
                      "volume": "883990.82612000",
                      "volume_30day": "57077184.26522",
                    },
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View>
                    <CellRenderer
                      cellKey="BTC-XXX"
                      fillRateHelper={
                        FillRateHelper {
                          "_anyBlankStartTime": null,
                          "_enabled": false,
                          "_getFrameMetrics": [Function],
                          "_info": Info {
                            "any_blank_count": 0,
                            "any_blank_ms": 0,
                            "any_blank_speed_sum": 0,
                            "mostly_blank_count": 0,
                            "mostly_blank_ms": 0,
                            "pixels_blank": 0,
                            "pixels_sampled": 0,
                            "pixels_scrolled": 0,
                            "sample_count": 0,
                            "total_time_spent": 0,
                          },
                          "_mostlyBlankStartTime": null,
                          "_samplesStartTime": null,
                        }
                      }
                      horizontal={false}
                      index={0}
                      inversionStyle={null}
                      item={
                        Object {
                          "high": "0.27933000",
                          "id": "BTC-XXX",
                          "last": "0.27020300",
                          "low": "0.26416200",
                          "open": "0.26416200",
                          "volume": "883990.82612000",
                          "volume_30day": "57077184.26522",
                        }
                      }
                      key="BTC-XXX"
                      onLayout={[Function]}
                      onUnmount={[Function]}
                      onUpdateSeparators={[Function]}
                      parentProps={
                        Object {
                          "data": Array [
                            Object {
                              "high": "0.27933000",
                              "id": "BTC-XXX",
                              "last": "0.27020300",
                              "low": "0.26416200",
                              "open": "0.26416200",
                              "volume": "883990.82612000",
                              "volume_30day": "57077184.26522",
                            },
                          ],
                          "disableVirtualization": false,
                          "getItem": [Function],
                          "getItemCount": [Function],
                          "horizontal": false,
                          "initialNumToRender": 10,
                          "keyExtractor": [Function],
                          "maxToRenderPerBatch": 10,
                          "numColumns": 1,
                          "onEndReachedThreshold": 2,
                          "removeClippedSubviews": false,
                          "renderItem": [Function],
                          "scrollEventThrottle": 50,
                          "updateCellsBatchingPeriod": 50,
                          "viewabilityConfigCallbackPairs": Array [],
                          "windowSize": 21,
                        }
                      }
                    >
                      <View
                        onLayout={[Function]}
                        style={null}
                      >
                        <View
                          onLayout={[Function]}
                          style={null}
                        >
                          <Item
                            item={
                              Object {
                                "high": "0.27933000",
                                "id": "BTC-XXX",
                                "last": "0.27020300",
                                "low": "0.26416200",
                                "open": "0.26416200",
                                "volume": "883990.82612000",
                                "volume_30day": "57077184.26522",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "backgroundColor": "rgb(255, 255, 255)",
                                  "marginBottom": 24,
                                  "padding": 10,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "rgb(255, 255, 255)",
                                    "marginBottom": 24,
                                    "padding": 10,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontSize": 14,
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "fontSize": 14,
                                        "fontWeight": "bold",
                                        "textAlign": "center",
                                      }
                                    }
                                  >
                                    BTC-XXX
                                  </Text>
                                </Text>
                                <Text>
                                  <Text>
                                    Open: 
                                    0.26416200
                                  </Text>
                                </Text>
                                <Text>
                                  <Text>
                                    High: 
                                    0.27933000
                                  </Text>
                                </Text>
                                <Text>
                                  <Text>
                                    Low: 
                                    0.26416200
                                  </Text>
                                </Text>
                                <Text>
                                  <Text>
                                    Volume: 
                                    883990.82612000
                                  </Text>
                                </Text>
                              </View>
                            </View>
                          </Item>
                        </View>
                      </View>
                    </CellRenderer>
                  </View>
                </View>
              </RCTScrollView>
            </_class>
          </ScrollViewMock>
        </VirtualizedList>
      </FlatList>
    </View>
  </View>
</ProductList>
`;
